crates/apub/src/activities/create_or_update/comment.rs

```diff
@@ -178,6 +178,17 @@ impl ActivityHandler for CreateOrUpdateNote {
     context: &Data<LemmyContext>,
     request_counter: &mut i32,
   ) -> Result<(), LemmyError> {
+    // Need to do this check here instead of Note::from_apub because we need the person who
+    // send the activity, not the comment author.
+    if let Some(true) = self.object.distinguished {
```

Comment:
This is wrong, it allows removing the distinguish flag from comments without any check. This would be so much cleaner if it was a separate activity.

Reply:
I agree, I probably shouldn't have done it like that in the first place. I'll make an issue for it, and do that sometime soon. #2708

Reply:
I was actually talking about federationt but its true for the API as well.